src/compiler/parser.rs

```diff
@@ -194,6 +234,37 @@ impl ParserType {
         node
     }
 
+    fn create_missing_node(
+        &mut self,
+        kind: SyntaxKind,
+        diagnostic_message: DiagnosticMessage,
+    ) -> MissingNode {
+        self.parse_error_at_current_token(&diagnostic_message);
+
+        let result = MissingNode::Identifier(self.factory.create_identifier(self, ""));
+        self.finish_node(result)
```

Comment:
...and since I implemented `NodeInterface` for `MissingNode` it can be passed to `finish_node()`